File: src/main/java/akko/ddbot/command/vLiverFinder.java
package akko.ddbot.command;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

import akko.ddbot.sql.SQLFun;
import akko.ddbot.sql.TwoTuple;
import cc.moecraft.icq.command.CommandProperties;
import cc.moecraft.icq.command.interfaces.GroupCommand;
import cc.moecraft.icq.event.events.message.EventGroupMessage;
import cc.moecraft.icq.sender.message.MessageBuilder;
import cc.moecraft.icq.user.Group;
import cc.moecraft.icq.user.GroupUser;

public class vLiverFinder implements GroupCommand {

    @Override
    public CommandProperties properties() {
        return new CommandProperties("vLiver");
    }

    @Override
    public String groupMessage(EventGroupMessage arg0, GroupUser arg1, Group arg2, String arg3,
            ArrayList<String> arg4) {
        try 
        {
            Class.forName("org.sqlite.JDBC");
            TwoTuple<ResultSet,Connection> tuple = new SQLFun().executeQuery("GroupInfo","SELECT * FROM vLiver;");
            ResultSet set = tuple.resultSet;
            MessageBuilder mb = new MessageBuilder();
            mb.add("----------").newLine();
            mb.add("uid | Name").newLine();
            while (set.next()) {
                mb.add(set.getString("vID") + "   " + set.getString("vNAME")).newLine();
            }
            tuple.connection.close();
            return mb.add("----------").toString();
        } 
        catch (ClassNotFoundException | SQLException e) {
            e.printStackTrace();
            return e.getMessage();
        }
    }
    
}

File: src/main/java/akko/ddbot/command/vLiverFinder.kt
package akko.ddbot.command

import akko.ddbot.BotMainActivity
import akko.ddbot.sql.KtormObject
import akko.ddbot.sql.connectionPool
import cc.moecraft.icq.command.CommandProperties
import cc.moecraft.icq.command.interfaces.GroupCommand
import cc.moecraft.icq.event.events.message.EventGroupMessage
import cc.moecraft.icq.sender.message.MessageBuilder
import cc.moecraft.icq.user.Group
import cc.moecraft.icq.user.GroupUser
import org.ktorm.database.Database
import org.ktorm.dsl.from
import org.ktorm.dsl.select
import java.sql.SQLException
import java.util.*

class vLiverFinder : GroupCommand {
    override fun properties(): CommandProperties {
        return CommandProperties("vLiver")
    }

    override fun groupMessage(arg0: EventGroupMessage, arg1: GroupUser, arg2: Group, arg3: String,
                              arg4: ArrayList<String>): String {
        return try {
            val connection = Database.connect(connectionPool.connectionPool)
            val mb = MessageBuilder()
            mb.add("----------").newLine()
            mb.add("uid | Name").newLine()
            for (row in connection.from(KtormObject.VliverInfo).select())
            {
                mb.add(row[KtormObject.VliverInfo.pid] + "   " + row[KtormObject.VliverInfo.name]).newLine()
            }
            mb.add("----------").toString()
        } catch (e: ClassNotFoundException) {
            BotMainActivity.ExceptionLogger!!.debug(e.message)
            e.printStackTrace()
            e.message!!
        } catch (e: SQLException) {
            BotMainActivity.ExceptionLogger!!.debug(e.message)
            e.printStackTrace()
            e.message!!
        }
    }
}